Dynatrace/dynatrace-operator
# Adds label-version-detection feature flag

# Description

Adds the `feature.dynatrace.com/label-version-detection: "true"` feature flag

If enabled, we add additional environment variables to pods injected with the oneagent,
these environment variables reference predefined label's values.

The label to env mapping:
- metadata.labels['app.kubernetes.io/version'] ==> `DT_RELEASE_VERSION`
- metadata.labels['app.kubernetes.io/part-of'] ==> `DT_RELEASE_PRODUCT`

Tried to come up with a code structure that makes sense for upcoming changes. (hence the TODOs)
- Did this so we don't have to rewrite stuff that much

## How can this be tested?
Have a  Dynakube that injects oneagents, add the feature flag.
- Deploy sample app, without the env vars set (should be added during injection)
- Deploy sample app, with the env vars already set (should be left alone during injection)


## Checklist
